src/sage/groups/misc_gps/argument_groups.py

```diff
@@ -1461,9 +1461,23 @@ def __pow__(self, exponent):
             1
             sage: S(-1)^3  # indirect doctest
             -1
+
+        Check that the results may live in other parents too::
+
+            sage: x = SR.var('x')
+            sage: elem = S(-1)^x; elem  # indirect doctest
+            (-1)^x
+            sage: elem.parent()
+            Symbolic Ring
+
         """
+        result = self._element_ ** exponent
         P = self.parent()
-        return P.element_class(P, self._element_ ** exponent)
+        try:
+            result = P.element_class(P, self._element_ ** exponent)
```

Comment:
why recompute the power ? and maybe use "return" in the try ?

Reply:
You are absolutely right, I changed the code in haste and missed replacing the argument. Fixed via c431e2cd15f.